internal/service/api/requests/get_permissions.go

```diff
@@ -0,0 +1,24 @@
+package requests
+
+import (
+	"net/http"
+
+	"gitlab.com/distributed_lab/kit/pgdb"
+	"gitlab.com/distributed_lab/logan/v3/errors"
+	"gitlab.com/distributed_lab/urlval"
+)
+
+type GetPermissionsRequest struct {
+	pgdb.OffsetPageParams
+
+	Link   *string `filter:"link"`
+	UserId *int64  `filter:"userId"`
+}
+
+func NewGetPermissionsRequest(r *http.Request) (GetPermissionsRequest, error) {
+	var request GetPermissionsRequest
+
+	err := urlval.Decode(r.URL.Query(), &request)
+
```

Comment:
As I remember .Wrap returns `nil` if original error is empty, so it's okay, BUT I just recommend to handle errors with if, cause it's easier to read code, moreover, some linters can swear about such approach and it will cause extra time to fix small things. But in this case its' okay, as you like more

Reply:
Yeah, `Wrap` returns `nil` if the error itself is nil, so I decided to utilize this feature to minimize the amount of code 😅 
It indeed looks confusing to people who are not aware it can return `nil`, so it probably makes sense to come back to `if`

Reply:
As you wish. if any problems in future, it can be fixed easily